main/lsp/requests/completion.h

```diff
@@ -10,9 +10,40 @@ class CompletionItem;
 class CompletionTask final : public LSPRequestTask {
     std::unique_ptr<CompletionParams> params;
 
+    // TODO(jez) Let's delete this?
     void findSimilarConstants(const core::GlobalState &gs, const core::lsp::ConstantResponse &resp, core::Loc queryLoc,
                               std::vector<std::unique_ptr<CompletionItem>> &items) const;
 
+    struct MethodSearchParams {
+        // TODO(jez) Remember: you should be able to construct a fake DispatchResult for <ErrorNode>
+        // constant responses.
+        std::shared_ptr<core::DispatchResult> dispatchResult;
+        size_t totalArgs;
+        bool isPrivateOk;
+    };
+
+    struct SearchParams {
+        // Point where completion results were requested
+        core::Loc queryLoc;
+
+        // Might be ""
+        std::string_view prefix;
+
+        // If nullopt, won't suggest methods.
+        std::optional<MethodSearchParams> forMethods;
+
+        // If false, won't suggest Ruby keywords
+        bool suggestKeywords;
+
+        // If not exists(), won't suggest locals for the enclosing method.
+        core::MethodRef enclosingMethod;
+
+        // If empty(), won't suggest constants.
+        core::lsp::ConstantResponse::Scopes scopes;
+    };
+    std::vector<std::unique_ptr<CompletionItem>> getCompletionItems(LSPTypecheckerDelegate &typechecker,
+                                                                    SearchParams params);
```

Comment:
What do you think about making params a constant reference? As it is, it will copy everything to pass it in to `getCompletionItems`.